src/mixins/raster-layer-cross-section-terrain-mixin.js

```diff
@@ -0,0 +1,325 @@
+  //
+  // It is not currently believed that Mesh2d marks need to support export, so the "getTransforms" method is not
+  // going to be included
+
+  const prop_descriptors = new Map()
+
+  prop_descriptors.set("x", new PositionChannelDescriptor("x"))
+  prop_descriptors.set("y", new PositionChannelDescriptor("y"))
+
+  prop_descriptors.set(
+    "longitude",
+    new GeographicChannelDescriptor("longitude", "x")
+  )
+
+  prop_descriptors.set(
+    "latitude",
+    new GeographicChannelDescriptor("latitude", "y")
+  )
+
+  prop_descriptors.set(
+    "strokeColor",
+    new StringPropDescriptor(
+      "strokeColor",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false
+    )
+  )
+  prop_descriptors.set(
+    "strokeWidth",
+    new NumericPropDescriptor(
+      "strokeWidth",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false,
+      is_zero_to_one
+    )
+  )
+  prop_descriptors.set(
+    "fillBelowLine",
+    new NumericPropDescriptor(
+      "fillBelowLine",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false,
+      is_zero_to_one
+    )
+  )
+
+  _layer.__genVega = function({
+    chart,
+    table,
+    filter,
+    lastFilteredSize,
+    globalFilter,
+    // eslint-disable-next-line no-unused-vars
+    pixelRatio,
+    layerName
+  }) {
+    const raster_layer_context = new RasterLayerContext(
+      chart,
+      table,
+      CrossSectionTerrainConfigDefinitionObject.key,
+      this,
+      layerName,
+      lastFilteredSize
+    )
+
+    _vega_property_output_state = materializePropDescriptors(
+      raster_layer_context,
+      prop_descriptors,
+      state
+    )
+    const {
+      sql_parser_transforms,
+      vega_data_formats,
+      vega_transforms,
+      vega_scales,
+      mark_properties
+    } = _vega_property_output_state.flatten()
+
+    assert(vega_data_formats.length <= 1)
+
+    if (typeof filter === "string" && filter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: filter
+      })
+    }
+
+    const post_filter_transform = create_post_filter_transform(
+      state.postFilters
+    )
+
+    if (post_filter_transform) {
+      sql_parser_transforms.push(post_filter_transform)
+    }
+
+    if (typeof globalFilter === "string" && globalFilter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: globalFilter
+      })
+    }
+
+    const data = [
+      {
+        name: layerName,
+        ...(vega_data_formats.length ? vega_data_formats[0] : {}),
+        sql: parser.writeSQL({
+          type: "root",
+          source: table,
+          transform: sql_parser_transforms
+        }),
+        enableHitTesting: state.enableHitTesting
+      },
+      ...vega_transforms
+    ]
+
+    const marks = [
+      {
+        type: CrossSectionTerrainConfigDefinitionObject.key,
+        from: {
+          data: layerName
+        },
+        properties: Object.assign({}, ...mark_properties)
```

Comment:
```suggestion
        properties: { ...mark_properties}
```

Reply:
So changing this is actually what caused the bug I mentioned, where the `mark` object was transformed into an array-like-object (basically all nested objects were proceeded with a stringified numeric index).  I'm honestly not too sure why, chatGPT gave me this highlighting the differences:

When using the object spread syntax ({ ...mark_properties}), it does not handle nested objects correctly. Instead of creating deep copies of the nested objects, it only performs a shallow copy, which leads to unexpected behavior when modifying the nested objects. This behavior is inherent to the object spread syntax and is known as shallow copying.

On the other hand, Object.assign({}, ...mark_properties) is more suitable for copying objects with nested structures. It performs a shallow copy of the top-level properties, but it recursively performs a shallow copy of the nested objects as well, ensuring that the resulting object maintains the correct structure.

```
const mark_properties = {
  prop1: 'value1',
  nestedObj: {
    prop2: 'value2',
  },
};

const spreadCopy = { ...mark_properties };
const assignCopy = Object.assign({}, mark_properties);

mark_properties.prop1 = 'new value';
mark_properties.nestedObj.prop2 = 'new value';

console.log(spreadCopy); // { prop1: 'value1', nestedObj: { prop2: 'new value' } }
console.log(assignCopy); // { prop1: 'value1', nestedObj: { prop2: 'value2' } }
```
I don't think this example really highlights what's going wrong here, but does give good indication that they are not equivalent expressions in javascript.  Going to leave it as Chris R originally had it.

Reply:
Sneaky! I glossed over the spread operator there, makes sense this is spreading all the objects within that object.